studio/components/interfaces/BillingV2/Usage/Activity.tsx

```diff
@@ -0,0 +1,269 @@
+import clsx from 'clsx'
+import ShimmeringLoader from 'components/ui/ShimmeringLoader'
+import SparkBar from 'components/ui/SparkBar'
+import { useDailyStatsQuery } from 'data/analytics/daily-stats-query'
+import { useProjectSubscriptionQuery } from 'data/subscriptions/project-subscription-query'
+import {
+  ProjectUsageResponse,
+  UsageMetric,
+  useProjectUsageQuery,
+} from 'data/usage/project-usage-query'
+import dayjs from 'dayjs'
+import { PRICING_TIER_PRODUCT_IDS } from 'lib/constants'
+import Link from 'next/link'
+import { Button, IconAlertTriangle } from 'ui'
+import { USAGE_APPROACHING_THRESHOLD } from '../Billing.constants'
+import BarChart from './BarChart'
+import SectionContent from './SectionContent'
+import SectionHeader from './SectionHeader'
+import { USAGE_CATEGORIES } from './Usage.constants'
+
+export interface ActivityProps {
+  projectRef: string
+}
+
+const Activity = ({ projectRef }: ActivityProps) => {
+  const { data: usage } = useProjectUsageQuery({ projectRef })
+  const { data: subscription } = useProjectSubscriptionQuery({ projectRef })
+  const { current_period_start, current_period_end } = subscription?.billing ?? {}
+  const startDate = new Date((current_period_start ?? 0) * 1000).toISOString()
+  const endDate = new Date((current_period_end ?? 0) * 1000).toISOString()
+  const categoryMeta = USAGE_CATEGORIES.find((category) => category.key === 'activity')
+
+  const upgradeUrl =
+    subscription?.tier.supabase_prod_id === PRICING_TIER_PRODUCT_IDS.ENTERPRISE
+      ? `/project/${projectRef}/settings/billing/enterprise`
+      : subscription?.tier.supabase_prod_id === PRICING_TIER_PRODUCT_IDS.TEAM
+      ? `/project/${projectRef}/settings/billing/team`
+      : `/project/${projectRef}/settings/billing/update`
+
+  const { data: mauData, isLoading: isLoadingMauData } = useDailyStatsQuery({
+    projectRef,
+    attribute: 'total_auth_billing_period_mau',
+    interval: '1d',
+    startDate,
+    endDate,
+  })
+
+  const { data: mauSSOData, isLoading: isLoadingMauSSOData } = useDailyStatsQuery({
```

Comment:
As the interval is for all usage stats, would make sense to have an endpoint that returns this data for all things usage-billing, then we'd avoid having to do so many individual requests, would also simplify this component quite a bit (not a blocker for this PR!)

Reply:
one consideration might also be that the payload might be really big and response times may slow down
the good thing about keep these as individual requests is that the charts aren't waiting for each other to be rendered

whichever requests completes first will be rendered - better perception of dashboard speed

Reply:
Good point, I'd definitely keep it like that in this PR, we can still revisit later to in case this is an issue or could be improved